novawalletTests/Common/Helpers/ChainFilterStrategyTests.swift

```diff
@@ -0,0 +1,365 @@
+            ChainModelGenerator.generateChain(assets: [], addressPrefix: 0)
+        ].reduceToDict()
+        
+        var resultChains: [ChainModel.Id: ChainModel] = [:]
+        
+        let updatedChains = currentChains.values.map { _ in
+            ChainModelGenerator.generateChain(
+                defaultChainId: generateChainId(notMatching: chainId),
+                generatingAssets: 0,
+                addressPrefix: 0
+            )
+        }
+        
+        let targetChainIdChains = [
+            ChainModelGenerator.generateChain(
+                defaultChainId: chainId,
+                generatingAssets: 0,
+                addressPrefix: 0
+            )
+        ]
+        
+        
+        let beforeFilterChanges = changes(updatedChains + targetChainIdChains)
+        
+        // when
+        let resultChanges = filterStrategy.filter(beforeFilterChanges, using: currentChains)
+        resultChains = resultChanges.mergeToDict(currentChains)
+        
+        let successCondition = resultChains[chainId] == nil
+            && resultChains.count == currentChains.count - targetChainIdChains.count
+        
+        // then
+        XCTAssert(successCondition)
+    }
+    
+    func testChainIdFilterInsert() {
+        chainIdFilterInserts { updatedChains in
+            updatedChains.map { DataProviderChange<ChainModel>.update(newItem: $0) }
+        }
+        
+        chainIdFilterInserts { updatedChains in
+            updatedChains.map { DataProviderChange<ChainModel>.insert(newItem: $0) }
+        }
+    }
+    
+    func chainIdFilterInserts(changes: ([ChainModel]) -> [DataProviderChange<ChainModel>]) {
+        // given
+        let chainId = Data.random(of: 32)!.toHex()
+        
+        let filterStrategy = ChainFilterStrategy.chainId(chainId)
+        
+        var resultChains: [ChainModel.Id: ChainModel] = [:]
+        
+        let newChains = [
+            ChainModelGenerator.generateChain(
+                assets: [],
+                defaultChainId: generateChainId(notMatching: chainId),
+                addressPrefix: 0
+            ),
+            ChainModelGenerator.generateChain(
+                assets: [],
+                defaultChainId: generateChainId(notMatching: chainId),
+                addressPrefix: 0
+            )
+        ]
+        
+        let targetChainIdChains = [ChainModelGenerator.generateChain(
+            defaultChainId: chainId,
+            generatingAssets: 0,
+            addressPrefix: 0
+        )]
+        
+        let beforeFilterChanges = changes(newChains + targetChainIdChains)
+        
+        // when
+        let resultChanges = filterStrategy.filter(beforeFilterChanges, using: [:])
+        resultChains = resultChanges.mergeToDict(resultChains)
+        
+        // then
+        XCTAssert(resultChains[chainId] != nil && resultChains.count == targetChainIdChains.count)
+    }
+    
+    func generateChainId(notMatching chainId: String) -> String {
```

Comment:
Don't think that this function makes sense in current implementation. There is very low possibility that chain ids will match given random function. So I would just go with ```Data.random```

Reply:
Tried to avoid any potential for test being flaky